### DISCUSSION TITLE
/usr/lib64/libldap.so.2: undefined reference to EVP_md2@OPENSSL_3.0.0

### DISCUSSION THREAD

**Discussion by umagmrit** [openssl/openssl#23639] [Q&A] [answered]
Getting below issue

OS : Rocky Linux 9

`/usr/lib64/libldap.so.2: undefined reference to EVP_md2@OPENSSL_3.0.0
`

```
$ strings /usr/lib64/libldap.so.2 | grep EVP_md2
EVP_md2
```

```
$ rpm -qf /usr/lib64/libldap.so.2
openldap-2.6.2-3.el9.x86_64
```

I have compiled `openssl 3.0.9` from source without `enable-md2` option . Do I need to use `enable-md2` to fix this issue ? is there any security concerns with `enable-md2` option ?

**Comment by t8m** [comment, marked as answer]
Yes, you need to add enable-md2 to fix this issue. Unless MD2 is actually used, there should be no issue.

**Comment by swplotner** [comment]
Would the original author be able to post what produced that error? I have an issue with Rocky Linux 9 and firedrake/icepack that used to work on Rocky Linux 8

**Reply by umagmrit** [reply]
I have faced this issue while compiling `zimbra-nginx` package on Rocky Linux 9.

checking for SASL library ... not found
checking for SASL library in /usr/local/ ... not found
checking for SASL library in /usr/pkg/ ... not found
checking for SASL library in /opt/local/ ... not found

./auto/configure: error: SASL support in the MAIL filter modules requires the SASL library.
You can either not enable the feature or install the libraries.

/bin/ld: /usr/lib64/libldap.so.2: undefined reference to EVP_md2@OPENSSL_3.0.0
collect2: error: ld returned 1 exit status

**Reply by tsuyopon** [reply]
Maybe, openssl specfile of Rocky version 9 needs "enable-md2" build option for built-in openldap package. 
https://git.rockylinux.org/staging/rpms/openssl/-/blob/r9/SPECS/openssl.spec?ref_type=heads#L311

This build option will avoid to append OPENSSL_NO_MD2 macro.
https://github.com/openssl/openssl/blob/openssl-3.0.9/Configure#L1812

If you do not add OPENSSL_NO_MD2 macro, EVP_md2 symbol add to openldap binary. (Example: openldap-2.6.6 is used as Rocky9.5 )
https://github.com/openldap/openldap/blob/OPENLDAP_REL_ENG_2_6_6/libraries/libldap/tls_o.c#L1064-L1066
